common/kotlinx-coroutines-core-common/src/CancellableContinuationImpl.kt

```diff
@@ -66,40 +67,63 @@ internal abstract class AbstractContinuation<in T>(
 
     internal val state: Any? get() = _state.value
 
-    public val isActive: Boolean get() = state is NotCompleted
+    public override val isActive: Boolean get() = state is NotCompleted
 
-    public val isCompleted: Boolean get() = state !is NotCompleted
+    public override val isCompleted: Boolean get() = state !is NotCompleted
 
-    public val isCancelled: Boolean get() = state is CancelledContinuation
+    public override val isCancelled: Boolean get() = state is CancelledContinuation
 
-    internal fun initParentJobInternal(parent: Job?) {
+    public override fun initCancellability() {
+        // This method does nothing. Leftover for binary compatibility with old compiled code
+    }
+
+    private fun installParentCancellationHandler() {
```

Comment:
we should check `if (isCompleted)` as the first action to avoid performance degradation in channels and mutex